packages/@lwc/engine-core/src/framework/template.ts

```diff
@@ -115,58 +115,77 @@ function validateLightDomTemplate(template: Template, vm: VM) {
+
+            let cacheKey = 0;
+            if (hasStyleToken && hasScopedStyles) {
+                cacheKey |= FragmentCache.HAS_SCOPED_STYLE;
+            }
+            if (hasStyleToken && isSyntheticShadow) {
+                cacheKey |= FragmentCache.SHADOW_MODE_SYNTHETIC;
+            }
 
-        if (!isUndefined(cache[cacheKey])) {
-            return cache[cacheKey];
-        }
+            if (!isUndefined(cache[cacheKey])) {
+                return cache[cacheKey];
+            }
 
-        const classToken = hasScopedStyles && hasStyleToken ? ' ' + stylesheetToken : '';
-        const classAttrToken =
-            hasScopedStyles && hasStyleToken ? ` class="${stylesheetToken}"` : '';
-        const attrToken = hasStyleToken && isSyntheticShadow ? ' ' + stylesheetToken : '';
-
-        let htmlFragment = '';
-        for (let i = 0, n = keys.length; i < n; i++) {
-            switch (keys[i]) {
-                case 0: // styleToken in existing class attr
-                    htmlFragment += strings[i] + classToken;
-                    break;
-                case 1: // styleToken for added class attr
-                    htmlFragment += strings[i] + classAttrToken;
-                    break;
-                case 2: // styleToken as attr
-                    htmlFragment += strings[i] + attrToken;
+            const classToken = hasScopedStyles && hasStyleToken ? ' ' + stylesheetToken : '';
+            const classAttrToken =
+                hasScopedStyles && hasStyleToken ? ` class="${stylesheetToken}"` : '';
+            const attrToken = hasStyleToken && isSyntheticShadow ? ' ' + stylesheetToken : '';
+
+            let htmlFragment = '';
+            for (let i = 0, n = keys.length; i < n; i++) {
+                switch (keys[i]) {
+                    case 0: // styleToken in existing class attr
+                        htmlFragment += strings[i] + classToken;
+                        break;
+                    case 1: // styleToken for added class attr
+                        htmlFragment += strings[i] + classAttrToken;
+                        break;
+                    case 2: // styleToken as attr
+                        htmlFragment += strings[i] + attrToken;
+                        break;
+                    case 3: // ${1}${2}
+                        htmlFragment += strings[i] + classAttrToken + attrToken;
+                        break;
+                }
             }
-        }
 
-        htmlFragment += strings[strings.length - 1];
+            htmlFragment += strings[strings.length - 1];
 
-        cache[cacheKey] = createFragment(htmlFragment);
+            cache[cacheKey] = createFragmentFn(htmlFragment);
 
-        return cache[cacheKey];
+            return cache[cacheKey];
+        };
     };
 }
 
+// Note: at the moment this code executes, createFragment have not being set.
+export const parseFragment = buildParseFragmentFn((html) => createFragment(html));
```

Comment:
this is weird... the first arrow function returns a fragment, while the second gets the fragment and then the first child is extracted, something is off, they should be symmetric. Also, I don't think any of these belongs here, it should be just in the DOM/Server package, it doesn't have to be called createFragment, could be something like `createStaticElement` or such...